viewer/static/js/app/components/containers/app-container.js

```diff
@@ -26,10 +26,12 @@ class AppContainer extends React.Component {
   }
 
   componentWillReceiveProps(nextProps) {
+    const previousDatasetId = this.datasetId;
```

Comment:
We have `store.datasetState.currentDataset.id` available that we might be able to connect to.